cli/spec/kontena/cli/stacks/deploy_command_spec.rb

```diff
@@ -43,6 +43,6 @@
     })
     expect(subject).to receive(:sleep).once
 
-    expect{subject.run(['test-stack'])}.to exit_with_error.and output(/deploy failed/).to_stderr
+    expect{subject.run(['test-stack'])}.to exit_with_error.and output(/Stack deploy failed/).to_stdout
```

Comment:
Errors should probably go to `$stderr`

Reply:
The way that `$stderr.puts` behaves when called in a spinner results in semi-broken output... `puts` gets queued up.

```
 [done] Creating stack affinity-fail      
 [done] Triggering deployment of stack affinity-fail     
    *   Waiting for deployment to start .. -Stack deploy failed
 - Cannot find applicable node for service instance development/affinity-fail/redis-1: Did not find any nodes for affinity filter: node==nonexist
 [fail] Waiting for deployment to start     
```

Reply:
Hmm right. There's some `spin.fail!` variant that does  something differently.

Reply:
The `wait_for_deployment_to_start` is called within a spinner, so it would need to take the spinner as an argument to use `spin.fail!`... 

```ruby
        spinner "Waiting for deployment to start" do
          wait_for_deployment_to_start(deployment)
        end
```

It would be easiest to just raise an exception with the entire multi-line message, but there's only support for handling `Kontena::Errors::StandardError`, which doesn't really fit here.

Reply:
If there was an error class such as `Error` then `raise Error, "dang!"` would get output like `Error : dang!` :) 

I don't know.. if you don't want to begin..rescue inside the spinner block, I guess putsing to stdout is good enough.

Of course spinner could do something like
```ruby
stderr_orig = $stderr
stderr_new = StringIO.new
$stderr = stderr_new

..do the spinning stuff..

$stderr = stderr_orig
stderr_new.rewind
$stderr.print sderr_new.read
```

to delay output of any stderr putsing to happen after the spinning has finished.

Reply:
I think using `puts` for spinner-compatible output together with the `abort` => `raise SystemExit` is the easiest thing to do here, without starting to refactor our error-handling infrastructure in the CLI.

I don't think adding explicit `begin ... rescues` to all spinner blocks is the right answer, and also capturing `$stderr` would break the use of `DEBUG=api` to see what's going on during spinners.